library.c

```diff
@@ -2023,6 +2021,7 @@ redis_read_xclaim_response(RedisSock *redis_sock, int count, zval *rv) {
 
             add_next_index_stringl(rv, id, li);
             efree(id);
+            id = NULL;
```

Comment:
I think er need to move this to a begin of `for` loop because there is `free` also in `else` block

Reply:
Good catch, moved.